internal/framework/status/conditionWithContextFunc_test.go

```diff
@@ -0,0 +1,94 @@
+package status_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/gomega"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	"github.com/nginxinc/nginx-gateway-fabric/internal/framework/controller/controllerfakes"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/framework/status"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/framework/status/statusfakes"
+)
+
+func TestConditionWithContextFunc_GetFails(t *testing.T) {
```

Comment:
The test content looks good! I think you can leverage subtests here though. 
```
tests := []struct {
    name string
    getReturns error
    updateReturns error
    expErr bool
    expRetry bool
} {{...}}
```

Reply:
Ok sure! I was thinking about using subtests but I was unsure if it would make it more difficult to understand and couldn't really think of a good way to do so given that we need to set the return stubs, but setting `getReturns` or `updateReturns` to `nil` and setting the stub on the fake with `nil` was something I never thought of.